packages/release-mastermind/src/contexts/pullRequests.ts

```diff
@@ -9,139 +1,262 @@
+/** @format */
+
+import * as core from "@actions/core"
+import { LoggingDataClass, LoggingLevels } from "@videndum/utilities"
+import { Context } from "vm"
+import { log } from ".."
+import { Config, PullRequestConfig, Release, Runners } from "../../types"
+import { CurContext, PRContext, Reviews, Version } from "../conditions"
+import { evaluator } from "../evaluator"
+import { Utils } from "../utils"
+import { Contexts } from "./methods"
 
 export class PullRequests extends Contexts {
-  context: PRContext
-  config: PullRequestConfig
-  constructor(
-    util: Utils,
-    runners: Runners,
-    configs: Config,
-    curContext: CurContext,
-    dryRun: boolean
-  ) {
-    if (curContext.type !== 'pr')
-      throw new LoggingDataClass(LoggingLevels.error, 'Cannot construct without issue context')
-    super(util, runners, configs, curContext, dryRun)
-    this.context = curContext.context
-    if (!configs.pr) throw new LoggingDataClass(LoggingLevels.error, 'Cannot start without config')
-    this.config = configs.pr
-  }
+	context: PRContext
+	config: PullRequestConfig
+	constructor(
+		util: Utils,
+		runners: Runners,
+		configs: Config,
+		curContext: CurContext,
+		dryRun: boolean
+	) {
+		if (curContext.type !== "pr")
+			throw new LoggingDataClass(
+				LoggingLevels.error,
+				"Cannot construct without pr context"
+			)
+		super(util, runners, configs, curContext, dryRun)
+		this.context = curContext.context
+		if (!configs.pr)
+			throw new LoggingDataClass(
+				LoggingLevels.error,
+				"Cannot start without config"
+			)
+		this.config = configs.pr
+	}
+
+	/**
+	 * Parse the PR Context
+	 * @author IvanFon, TGTGamer, jbinda
+	 * @since 1.0.0
+	 */
+	static async parse(
+		utils: Utils,
+		config: Config,
+		context: Context
+	): Promise<PRContext | undefined> {
+		const pr = context.payload.pull_request
+		if (!pr) {
+			return
+		}
+
+		log(
+			LoggingLevels.debug,
+			`context.payload.pull_request: ` +
+				JSON.stringify(context.payload.pull_request)
+		)
+
+		const IDNumber = pr.number
+		const labels = await utils.parsingData.labels(pr.labels).catch((err) => {
+			log(LoggingLevels.error, `Error thrown while parsing labels: `, err)
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Videndum/action-masterminds/pullRequest?prid=7872005)
```suggestion
      log(LoggingLevels.error, "Error thrown while parsing labels: ", err)
```